tests/test_steps_edge.py

```
@@ -8,97 +15,107 @@
+    Ensures project data is cleared but the warm_start_cache is preserved.
+    """
     df_orig = pd.DataFrame({"Name": ["P1"], config.COL_GROUP: [1], "S1": [10.0]})
     mock_state = MagicMock()
+    # Simulate existing state
+    mock_state.keys.return_value = ["participants_df", "warm_start_cache"]
     mock_state.interactive_df = df_orig
+
     with patch("src.ui.steps.st") as mock_st:
         mock_st.session_state = mock_state
-        # Force the 'Start Over' button branch
         mock_st.button.side_effect = lambda label, **kwargs: label == "🔄 Start Over"
```

Ruff (0.15.12)

[warning] Unused lambda argument: `kwargs`

(ARG005)